packages/ra-core/src/form/getSimpleValidationResolver.ts

```diff
@@ -50,26 +50,80 @@ export const getSimpleValidationResolver = (validate: ValidateForm) => async (
 ) => {
     const errors = await validate(data);
 
-    if (!errors || Object.getOwnPropertyNames(errors).length === 0) {
+    // If there are no errors, early return the form values
+    if (!errors || isEmptyObject(errors)) {
         return { values: data, errors: {} };
     }
-    const flattenedErrors = flattenErrors(errors);
 
+    // Else, we return an error object shaped like errors but having for each leaf
+    // `type: 'manual'` and a `message` prop like react-hook-form expects it
+    const transformedErrors = transformErrorFields(errors);
+
+    // If, after transformation, there are no errors, return the form values
```

Comment:
it can happen if the validator returns an error object like:

```js
        {
            backlinks: [{}, {}],
        }
```

in this case we first need to transform the error field to realize it actually contains no errors.

I'll add this in the comment.